include/mbedtls/aes.h

```diff
@@ -67,6 +67,49 @@
 /* MBEDTLS_ERR_AES_HW_ACCEL_FAILED is deprecated and should not be used. */
 #define MBEDTLS_ERR_AES_HW_ACCEL_FAILED                   -0x0025  /**< AES hardware accelerator failed. */
 
```

Comment:
I think we should take the chance to document all parameter restrictions explicitly. For example:
```
 * \brief          This function initializes the specified AES context.
 *
 *                 It must be the first API called before using
 *                 the context.
 *
 * \param ctx      The AES context to initialize.
 */
void mbedtls_aes_init( mbedtls_aes_context *ctx );
```
could become
```
 * \brief          This function initializes the specified AES context.
 *
 *                 It must be the first API called before using
 *                 the context.
 *
 * \param ctx      The AES context to initialize. This is assumed to be not 
 *                 \c NULL if #MBEDTLS_CHECK_PARAMS is unset (default).
 */
void mbedtls_aes_init( mbedtls_aes_context *ctx );
```

Reply:
I'd go for something much simpler:
```
 * \param ctx      The AES context to initialize.  Must not be \c NULL.
```
My reasoning is as follows:

- "must not" is explicit enough
- we're going to add this to a lot of places so we don't want it to be too long (with the risk of people missing other important bits of the documentation because they're lost in the middle of repetitive stuff)
- let's avoid duplication, and centralize the documentation of what `MBEDTLS_CHECK_PARAMS` does in a single place